api/types/types.go

```diff
@@ -54,13 +54,37 @@ type ImageMetadata struct {
 	LastTagTime time.Time `json:",omitempty"`
 }
 
+// ImagePlatform describes the platform which the image in the manifest runs on.
+// This type is our equivalent of ocispec.Platform, with Capitalized JSON field names.
```

Comment:
Question: the `/distribution/{name}/json` endpoint already returns platform information, but that's a direct use of the ocispec.Platform, which means that that endpoint does not use the same capitalisation.

Should we omit this type (and use a direct copy of the OCI-spec, or change the other endpoint?

In general, I prefer that we define our own types for the API (in either case), instead of types we don't (directly) control (to prevent inadvertently changing API responses)

Reply:
Hmm, it seems strange IMO for us to be returning what's effectively an "OCI Platform" object without using the canonical capitalization for the fields, so I'd vote anywhere we return a full proper "platform object" that it be the lowercase OCI versions.

Reply:
(I don't have a strong opinion one way or another on using the OCI structs directly.)

Reply:
Same. We can use our own types if desired but it would be nice to use the industry standard field name format.

Reply:
Oh! Thought I pushed 🤦 

I updated to use the same casing as upstream; still using our own type (with the addition of a Stringer() interface), but otherwise same as the OCI type